crypto/evp/evp_enc.c

```diff
@@ -934,40 +942,50 @@ int EVP_CIPHER_CTX_set_key_length(EVP_CIPHER_CTX *c, int keylen)
 
 int EVP_CIPHER_CTX_set_padding(EVP_CIPHER_CTX *ctx, int pad)
 {
+    int ok;
     if (pad)
         ctx->flags &= ~EVP_CIPH_NO_PADDING;
     else
         ctx->flags |= EVP_CIPH_NO_PADDING;
 
-    if (ctx->cipher != NULL && ctx->cipher->prov != NULL) {
-        OSSL_PARAM params[] = {
-            OSSL_PARAM_int(OSSL_CIPHER_PARAM_PADDING, NULL),
-            OSSL_PARAM_END
-        };
-
-        params[0].data = &pad;
-
-        if (ctx->cipher->ctx_set_params == NULL) {
-            EVPerr(EVP_F_EVP_CIPHER_CTX_SET_PADDING, EVP_R_CTRL_NOT_IMPLEMENTED);
-            return 0;
-        }
-
-        if (!ctx->cipher->ctx_set_params(ctx->provctx, params))
-            return 0;
-    }
-
-    return 1;
+    ok = evp_do_param(ctx->cipher, &pad, sizeof(pad),
+                      OSSL_CIPHER_PARAM_PADDING, OSSL_PARAM_INTEGER,
+                      evp_do_ciph_ctx_setparams, ctx->provctx);
+    return ok != 0 ? 1 : 0;
```

Comment:
As @paulidale reminded me, you don't need the ```?...``` part, the logical test sets the right 0 or 1 value. and then just do this:
```
return evp_do_param(....) != 0;
```

Reply:
So...  like this?

``` C
    return evp_do_param(ctx->cipher, &pad, sizeof(pad),
                        OSSL_CIPHER_PARAM_PADDING, OSSL_PARAM_INTEGER,
                        evp_do_ciph_ctx_setparams, ctx->provctx) != 0;
```

That's not exactly readable, so maybe this?

``` C
    return
        evp_do_param(ctx->cipher, &pad, sizeof(pad),
                     OSSL_CIPHER_PARAM_PADDING, OSSL_PARAM_INTEGER,
                     evp_do_ciph_ctx_setparams, ctx->provctx)
        != 0;
```

Clearer, but a bit on the ugly side, wouldn't you say?  Anyway, if you insist,I'll do the latter.

Reply:
I don't have a vote so I don't insist :)

Reply:
At the very least, I can change `ok != 0 ? 1 : 0` to `ok != 0`.

Reply:
I'd suggest leaving the function call alone and having `return ok != 0`